tooling/nargo/src/errors.rs

```diff
@@ -45,6 +45,9 @@ pub enum NargoError {
     /// Oracle handling error
     #[error(transparent)]
     ForeignCallError(#[from] ForeignCallError),
+
+    #[error("Failed to solve ACIR function")]
+    AcirCallError { id: u32, error: Box<NargoError>, location: OpcodeLocation },
```

Comment:
Do we really need this? In my mind this is just an execution layer but we need to be able to tell which circuit we're in when the error occurs. Why not just add the circuit id into the `ExecutionError` variant?

Reply:
I potentially could do that I guess, but then I would have to include that id inside every variant as well no? And not just for `ExecutionError` but also `ForeignCallError`? This felt more complex as `execute_circuit` returns a `NargoError` which also included a `ForeignCallError`

Reply:
I think that it's conceptually simpler and more correct though. Why should I get a different error when a constraint is broken in a called acir function vs the main function? We're also going to be getting nested errors with `AcirCallError`s inside `AcirCallErrors`, etc. which we need to deal with, etc.

The proper solution is to augment `OpcodeLocation` (probably not by modifying this struct but we need to pass this data at the same time) so that it properly specifies the location which threw the error.

Reply:
> The proper solution is to augment `OpcodeLocation` (probably not by modifying this struct but we need to pass this data at the same time) so that it properly specifies the location which threw the error.

I went with a new `ResolvedOpcodeLocation` struct. Let me know what you think, but I agree that it is much cleaner than the recursive `NargoError`.